python/lsst/ap/association/dia_object.py

```diff
@@ -175,24 +206,10 @@ def append_dia_source(self, input_dia_source):
         # them until we are finished adding sources.
         self._updated = False
 
-        # Do stuff to append this SourceRecord.
+        self._dia_source_catalog.append(input_dia_source_record)
```

Comment:
Currently, this does not copy the source. This will be taken care of by the update method later which deep copies the full catalog. Would it be enough to add a comment to the docstring that the source isn't properly copied until update is run?